### PR TITLE
[release_2.0] Work around kernel bug that forces fuse-overlayfs in podman (#765)

### PR DESCRIPTION
**Pull request by samdoran**
Backport of #765 for Ansible Runner 2.0.

### CODE REVIEW

**Comment by samdoran** [comment]
recheck

**Comment by Shrews** [comment]
recheck

File: test/unit/config/test__base.py [hunk 1]
```diff
@@ -314,3 +314,28 @@ def test_containerization_settings(tmpdir, container_runtime):
             assert '--user=' in rc.command[index]
         else:
             assert rc.command[index] == element
+
+
+@pytest.mark.parametrize(
+    'container_runtime, expected', (
+        ('docker', None),
+        ('podman', '1')
```
Comment (Shrews) [inline test/unit/config/test__base.py:322]: We really need to just fix it so that these tests are actually run rather than doing this awkward mock.

Comment (samdoran) [reply test/unit/config/test__base.py:322]: Or refactor the code so we can pass in a parameter for testing use rather than having to do all this setup.

**Pull request merged**